### PR TITLE
Refactor parts of ember-app and minor cleanups

### PR DESCRIPTION
**Pull request by kratiahuja**
This PR refactors some of the functionality of ember-app into smaller functions. There is no functionality change but aims to keep functionality together and independent. It also cleans up unneeded functions and packages.

cc: @tomdale @danmcclain @arjansingh @masonwan @ryanone

### CODE REVIEW

**Comment by kratiahuja** [comment]
We need this temporarily until this issue is [fixed](https://github.com/ember-cli/ember-cli/issues/6350) and some other issues are fixed. See this #91 for additional details.

File: src/ember-app.js [hunk 1]
```diff
@@ -177,6 +200,39 @@ class EmberApp {
...
+        instance = appInstance;
+        result.instance = instance;
+        registerFastBootInfo(fastbootInfo, instance);
+
+        return instance.boot(bootOptions);
+      })
+      .then(() => result.instanceBooted = true)
+      .then(() => instance.visit(path, bootOptions))
+      .then(() => waitForApp(instance))
+      .then(() => {
```
Comment (arjansingh) [inline src/ember-app.js]: nitpicking, with ES6 this could just be:
`.then(() => instance);`

**Comment by kratiahuja** [comment]
@rwjblue This is rebased as well. But I think I will need to rebase again once PR #92 is merged.

**Comment by kratiahuja** [comment]
Rebased and ready to merge now.

**Comment by tomdale** [comment]
Looks like a nice improvement, thank you @kratiahuja!

**Pull request merged**